corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -153,42 +176,69 @@ hqDefine("geospatial/js/geospatial_map", [
                 return map;
             };
 
-            self.removeMarkers = function () {
-                // Remove markers
-                _.each(caseModels(), function(currCase) {
+            self.removeCaseMarkers = function () {
+                _.each(caseModels(), function (currCase) {
                     if (currCase.marker) {
                         currCase.marker.remove();
                     }
                 });
                 caseModels([]);
             };
 
+            self.removeUserMarkers = function () {
+                _.each(userModels(), function (currUser) {
+                    if (currUser.marker) {
+                        currUser.marker.remove();
+                    }
+                });
+                userModels([]);
+            };
+
             self.addCaseMarkersToMap = function (rawCases) {
-                _.each(rawCases, function(element) {
-                    let coordinates = element.coordinates;
+                const caseColors = {
+                    'default': defaultCaseMarkerColor,
+                    'selected': selectedCaseMarkerColor,
+                };
+
+                _.forEach(rawCases, function (element, caseId) {
+                    const coordinates = element.coordinates;
                     if (coordinates && coordinates.lat && coordinates.lng) {
-                        self.addMarker(element, defaultMarkerColor);
+                        const mapItemInstance = self.addMarker(caseId, element, caseColors);
+                        caseModels.push(mapItemInstance);
                     }
                 });
             };
 
-            self.addMarker = function (rawCase, color) {
-                let coordinates = rawCase.coordinates;
+            self.addUserMarkersToMap = function (rawUsers) {
```

Comment:
can we make a common function for this and `addCaseMarkersToMap`?

Reply:
[ef300ef](https://github.com/dimagi/commcare-hq/pull/33488/commits/ef300ef11e125ef447fd01d1762338d3ec6cbf52)